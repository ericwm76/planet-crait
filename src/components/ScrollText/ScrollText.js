import React from 'react';
import './ScrollText.scss';

const ScrollText = (props) => {
    const {episode_id, opening_crawl, title} = props.movie;
    return (
        <section className='scroll-text'>
            <div className="fade"></div>
          <section className='star-wars'>
              <div className='crawl'>
                <div className='title'>
                   <p>Episode {episode_id} </p>
                   <h3>{title}</h3>
                </div>
                {opening_crawl}
              </div>
          </section>
        </section>
    )
}

export default ScrollText;