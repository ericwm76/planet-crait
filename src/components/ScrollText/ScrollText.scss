.scroll-text {
  background-image: url('https://wallpaperfm.com/img/original/3/5/2/16842.jpg');
  height: 100vh;
  overflow: hidden;
  background-repeat: no-repeat;
}

.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.fade {
    position: relative;
    width: 100%;
    min-height: 60vh;
    top: -25px;
    background-image: linear-gradient(0deg, transparent, black 75%);
    z-index: 1;
  }

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
    0% {
        top: 0;
        transform: rotateX(20deg) translateZ(0);
    }
    100% {
        top: -6000px;
        transform: rotateX(25deg) translateZ(-2500px);
    }
}