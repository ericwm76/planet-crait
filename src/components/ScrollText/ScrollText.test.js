import React from 'react';
import { shallow } from 'enzyme';
import ScrollText from './ScrollText';

describe('ScrollText', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ScrollText />)
  })

  it('should match snapshot with all data passing in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})