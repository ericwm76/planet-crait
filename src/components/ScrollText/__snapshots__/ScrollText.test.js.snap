// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText should match snapshot with all data passing in correctly 1`] = `
<section
  className="scroll-text"
/>
`;
